### PR TITLE
Add Asus settings page and add some features

### PR DESCRIPTION
**Pull request by Rocky7842**
* Add a new toggle
* Also include method for zf8 according to (c00b741)

### CODE REVIEW

**Comment by phhusson** [comment]
Thanks, but unless you plan to add other Asus features, I'd rather you keep the DT2W option inside Misc

**Comment by Rocky7842** [comment]
> Thanks, but unless you plan to add other Asus features, I'd rather you keep the DT2W option inside Misc

Also found two persist props that may be added, from my ROG Phone 1.

1. `persist.asus.fp.wakeup=true`
These are used to enable the fingerprint sensor when the screen is off.
(Probably available on devices with fp sensor on the back.)
2. `persist.vendor.usb.controller.default=a800000.dwc3` (or a600000.dwc3 here)
This is used to select which usb port to be used for data transfer.
(Probably available on ROG Phones.)

But I haven't yet validated how common these two props are. I will check it and update accordingly.

**Comment by phhusson** [comment]
ok, then please add those in this PR. FYI you can list the available controllers with `ls /sys/class/udc`

**Comment by Rocky7842** [comment]
* Add Asus settings page
* Add DT2W, including method for zf8 according to (c00b741)
* Add glove mode
* Add fingerprint wake
* Add USB port picker

**Comment by Rocky7842** [comment]
> All good for me. Just to confirm, `val process = Runtime.getRuntime().exec("ls /sys/class/udc")` worked without needing to change any selinux policy or anything?

Forgot I was in permissive. Take a look at logs and get these:
`ls      :type=1400 audit(0.0:6906): avc: denied { read } for name="udc" dev="sysfs" ino=40203 scontext=u:r:system_app:s0 tcontext=u:object_r:sysfs:s0 tclass=dir`
`ls      : type=1400 audit(0.0:6910): avc: denied { open } for path="/sys/class/udc" dev="sysfs" ino=40203 scontext=u:r:system_app:s0 tcontext=u:object_r:sysfs:s0 tclass=dir`

Looks like we indeed need to change something, but I don't really know much about sepolicy. Can you help me with this?

**Comment by phhusson** [comment]
Granting full sysfs access to system_app is really not great.
I'd rather you add something like `setprop sys.usb.all_controllers "$(ls /sys/class/udc |tr ' ' ',')"` in system/bin/rw-system.sh

**Comment by Rocky7842** [comment]
> Granting full sysfs access to system_app is really not great. (and changing sepolicy of /sys/class/udc sounds like a good idea to break USB) I'd rather you add something like `setprop sys.usb.all_controllers "$(ls /sys/class/udc |tr ' ' ',')"` in system/bin/rw-system.sh

Ok.

**Comment by Rocky7842** [comment]
With changes pushed to device_phh_treble, everything is now working with selinux enforced.

File: app/src/main/res/xml/pref_asus.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            android:key="key_asus_glove_mode"
+            android:title="Enable glove mode" />
+        <SwitchPreference
+            android:defaultValue="false"
+            android:key="key_asus_dt2w"
+            android:title="Enable double tap to wake" />
+    </PreferenceCategory>
+    <PreferenceCategory android:title="Other">
+        <SwitchPreference
+            android:defaultValue="false"
```
[the remark this answers is not in the document]
Comment (Rocky7842) [reply app/src/main/res/xml/pref_asus.xml:15]: > Isn't the default behavior "true"? I mean the vast majority of smartphones has it to true by default and can't be disabled

Double checked and yes, most of the devices are default to true, including mine. I have also found a few devices default to false, but it's `persist.asus.fp.wakeup_support` is true, so it should be supported anyways.
Note: This prop doesn't do anything from my test, so I didn't include it here.

[the remark this answers is not in the document]
Comment (Rocky7842) [reply app/src/main/res/xml/pref_asus.xml:15]: Also, I found these during the research of stock dumps. These are from the rog phone 2.
`on property:persist.asus.dclick=1`
`    write sys/devices/platform/goodix_ts.0/gesture/dclick 0`
`on property:persist.asus.glove=1`
`   write sys/devices/platform/goodix_ts.0/glove_mode 1`
`on property:persist.asus.fp.wakeup=true`
`   write sys/devices/platform/goodix_ts.0/gesture/aod_enable 1`
Should I also add support for these sysfs node?

**Comment by Rocky7842** [comment]
I also found these during the research of stock dumps. These are from the rog phone 2.
on property:persist.asus.dclick=1
    write sys/devices/platform/goodix_ts.0/gesture/dclick 1
on property:persist.asus.glove=1
   write sys/devices/platform/goodix_ts.0/glove_mode 1
on property:persist.asus.fp.wakeup=true
   write sys/devices/platform/goodix_ts.0/gesture/aod_enable 1
Should I also add support for these sysfs nodes?

**Comment by Rocky7842** [comment]
Also found those props from the stock dump of ROG Phone 2, so I removed the misleading summary for fp wake.

**Pull request merged**